e2e-tests/playwright/specs/functional/system_console/permissions/team_access.spec.ts

```diff
@@ -0,0 +1,176 @@
+// final API call that are all ultimately part of the spec, and we want to test the effects of those, not
+// merely a hand-created version of the same.
+const setupDefaultSystemManagerRole = async (
+    pw: PlaywrightExtended,
+    adminUser: UserProfile,
+    sectionTestId: string,
+    subsectionTestId: string,
+    permissionText: string,
+) => {
+    // Login as admin and navigate to System Console
+    const {systemConsolePage: adminConsolePage} = await pw.testBrowser.login(adminUser);
+
+    // Go to System Console
+    await adminConsolePage.goto();
+
+    // Login to the System Console and navigate to Delegated Granular Administration
+    await adminConsolePage.sidebar.goToItem('Delegated Granular Administration');
+
+    // Find the System Manager row in the table
+    const systemManagerText = adminConsolePage.page.getByText('System Manager', {exact: true}).first();
+    await expect(systemManagerText).toBeVisible();
+
+    // Click on the System Manager text to go to its settings page
+    await systemManagerText.click();
+
+    // Expand the section
+    const sectionReporting = adminConsolePage.page.locator(`data-testid=${sectionTestId}`);
+    const hideSubsectionsLink = sectionReporting.getByRole('button').filter({hasText: 'Hide'}).first();
+    const showSubsectionsLink = sectionReporting.getByRole('button').filter({hasText: 'Show'}).first();
+
+    // Check which one is visible and click if needed
+    const isHideVisible = await hideSubsectionsLink.isVisible();
+    const isShowVisible = !isHideVisible && (await showSubsectionsLink.isVisible());
+
+    if (isShowVisible) {
+        // Need to expand
+        await showSubsectionsLink.click();
+    }
+
+    // Get the whole row
+    const rowReporting = adminConsolePage.page.locator('.PermissionRow').filter({has: sectionReporting});
+    await rowReporting.click();
+
+    // Find the sub section
+    const subsectionTeamStatistics = rowReporting.locator(`data-testid=${subsectionTestId}`);
+
+    await subsectionTeamStatistics.click();
+
+    // Look for dropdown button
+    const dropdownButton = subsectionTeamStatistics.locator('button').first();
+
+    // Click the button to open the dropdown menu
+    await dropdownButton.click();
+
+    // Click on the desired option in the dropdown
+    const permissionOption = subsectionTeamStatistics.locator('.dropdown-menu').getByText(permissionText).first();
+    await permissionOption.click();
+
+    // Click Save button
+    const saveButton = adminConsolePage.page.getByRole('button', {name: 'Save'}).first();
+    await saveButton.click();
+
+    // Wait for save operation to complete
+    await adminConsolePage.page.waitForLoadState('networkidle');
+
+    // Go back to the main console
+    await adminConsolePage.goto();
+    await adminConsolePage.page.waitForLoadState('networkidle');
+};
+
+test('MM-63378 System Manager without team access permissions cannot view team details', async ({pw}) => {
```

Comment:
Please send a PR to the test management repo, so we can save the test case and have it assigned with correct test key.

Reply:
Ah, hmm. What can we do to automate or eliminate this additional process? cc @streamer45

Reply:
For now, that's what we have but I have filed a ticket (https://mattermost.atlassian.net/browse/MM-63754) to automate the process. On that ticket/task:

What will remain
- test case to be saved to central place (Zephyr)
- test case will require review

Process to change
- review the test case where the code change is submitted (first with Playwright)
- automate saving to Zephyr (new GH action/job)

Reply:
@saturninoabril, I don't see any feature developer contributions to https://github.com/mattermost/mattermost-test-management over the past year. How are existing tests getting into this repository today?

Reply:
Most QA folks still create and update test cases directly in the Zephyr UI. Those changes get synced to the repo during the data sync. There’s a bit of discussion about it here - https://github.com/mattermost/mattermost-test-management/pull/205

Since adoption is low and it creates additional step/process, it's about time to think about a better way.